### DISCUSSION TITLE
Drawing

### DISCUSSION THREAD

**Discussion by ghost** [Milkdown/milkdown#224] [Ideas]
Is it feasible to create a drawing plugin that matches (or supersedes) that of [Tiptap](https://tiptap.dev/examples/drawing)?

It doesn't necessarily have to be better, but just "drawable". ^^

**Comment by ghost** [comment]
I was thinking in something like inserting an SVG canvas and when exported (CTRL + C), write to a blob.

**Reply by ghost** [reply]
They're using d3 to draw.

**Comment by Saul-Mirone** [comment]
Integrate a draw component won't be difficult, but it's hard to make it transform between markdown strings.

**Comment by ghost** [comment]
I was thinking in two simple scenarios:
- when copying, transform to data image URL;
- when pasting, well, it's just another image, no conversion is made.

Much like [notekit](https://github.com/blackhole89/notekit/) handles drawing.

With that, the work could be done in a worker thread, with a small web assembly module, like the one [thumbo](https://www.npmjs.com/package/thumbo) uses. It's lightweight and open source! Then there would be 2 direct interfaces: (1) for the core plugin to send the data url **as SVG** through (2) the webassembly module to make the conversion to PNG as data url, and send it back to "copy" event.

**Reply by ghost** [reply]
There are [other ways](https://stackoverflow.com/questions/50478133/d3js-export-svg-to-image) to export to PNG, but I personally dislike the idea of having deep dependencies.
